libraries/AP_Terrain/AP_Terrain.cpp

```diff
@@ -215,16 +215,11 @@ bool AP_Terrain::height_above_terrain(float &terrain_altitude, bool extrapolate)
     if (!height_terrain_difference_home(terrain_difference, extrapolate)) {
         return false;
     }
-    Location loc;
-    if (!ahrs.get_position(loc)) {
```

Comment:
> And again I believe every caller locks a home before EKF can lock an origin, so it should still be fine

I'm pretty sure it's the other way around. If the EKF couldn't lock an origin how could the vehicle lock a home?

This is repetitive (sorry) but the trouble here is *get_relative_position_D_home* not returning if it is valid or not and just assuming that baro altitude is the altitude above home. If it was returning a boolean then we would check here if we had a valid altitude above home or not.

I also don't agree we need to check the location at this point in code. The current location is needed in the method called above, *height_terrain_difference_home* and it is checked there. If we got this far we only need the altitude above home.

All things said, I think this is a good change.